stages/org.osbuild.copy

```diff
@@ -54,6 +62,11 @@ SCHEMA_2 = r"""
                 "pattern": "^tree:\/\/\/"
               }
             ]
+          },
+          "remove_destination": {
+            "type": "boolean",
+            "description": "Remove the destination before copying. Works only for files, not directories.",
```

Comment:
Since we specify that `remove_destination` only works for directories, maybe we should error out if the destination is a directory in the code ?

Reply:
If the destination is a directory, then the file will be copied into it. This is compliant with how `cp` bahaves. I added the comment about the bahavior if the target is a directory, because I think that it is important to note. However I would not consider using the option with directories to be a hard error. Osbuild is not intended for direct use by end users and IMO anyone who decides to use it directly should be able to read the behavior of stages that they use. Nevertheless I'm open to making this situation a hard error, if there is a wider consensus on it. Although I'm not in favor of it.

Reply:
That makes sense. It makes me feel like it would be nice to have warnings in OSBuild stages. This is a situation where they would make sense.